### PR TITLE
Expand feature_flags macro to accept default and internal flag

### PR DESCRIPTION
**Pull request by antiguru**
This extends the `feature_flags` macro to accept a default value, and the internal flag. With this change, the macro should be usable in more situations than it currently is.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by sploiselle — COMMENTED** [review]
No qualms with the change itself, but would love to improve the macro's docs.

File: src/sql/src/session/vars.rs [hunk 1]
```diff
@@ -1257,26 +1257,35 @@ mod cluster_scheduling {
 ///
 /// Note that not all `ServerVar<bool>` are feature flags. Feature flags are for variables that:
 /// - Belong to `SystemVars`, _not_ `SessionVars`
-/// - Default to false and must be explicitly enabled
+/// - Default to false and must be explicitly enabled or provide an optional value and internal flag.
 macro_rules! feature_flags {
-    ($(($name:expr, $feature_desc:literal)),+ $(,)?) => {
+    (@inner $name:expr, $feature_desc:literal,) => {
```
[the remark this answers is not in the document]
Comment (antiguru) [reply src/sql/src/session/vars.rs]: Done.

File: src/sql/src/session/vars.rs [hunk 2]
```diff
@@ -1257,26 +1257,35 @@ mod cluster_scheduling {
 ///
 /// Note that not all `ServerVar<bool>` are feature flags. Feature flags are for variables that:
 /// - Belong to `SystemVars`, _not_ `SessionVars`
-/// - Default to false and must be explicitly enabled
+/// - Default to false and must be explicitly enabled or provide an optional value and internal flag.
```
[the remark this answers is not in the document]
Comment (antiguru) [reply src/sql/src/session/vars.rs]: Updated the documentation.

**Comment by antiguru** [comment]
Thanks for the review, updated the documentation.

**Review by sploiselle — APPROVED** [review]

**Pull request merged**